src/merge_iterator.rs

```diff
@@ -135,11 +141,22 @@ impl<T1: KeyValueIterator, T2: KeyValueIterator> KeyValueIterator for TwoMergeIt
             if peeked_kv.key != current_kv.key {
                 break;
             }
-            if peeked_kv.seq > current_kv.seq {
+            if peeked_kv.seq > current_kv.seq
+                && self
+                    .max_seq
+                    .map(|max_seq| peeked_kv.seq <= max_seq)
+                    .unwrap_or(true)
+            {
                 current_kv = peeked_kv.clone();
             }
             self.advance_inner().await?;
         }
+
+        if let Some(max_seq) = self.max_seq {
```

Comment:
I suppose this is needed in case the very first element in the iterator has a seq > max_seq? If so, WDYT about putting this logic in the `let mut current_kv = match self.advance_inner().await? {` block above? Something like.. `Some(kv) if kv.seq <= self.max_seq` (though would need to unwrap max_seq).

Reply:
yes indeed, this is needed on the case which the very first element's seq is bigger than max_seq. i like this style you suggested, let me have a try on this :+1:.

Reply:
oh sorry the `current_key` at the beginning is still a must, because the loop in the merge logic will break as soon as the peeked key is not the current key, a 'merge' only focuses on a single key's different versions, so it requires to know which key is the 'current key'.